gms/gossiper.hh

```diff
@@ -209,6 +210,8 @@ private:
     /* unreachable member set */
     std::unordered_map<inet_address, clk::time_point> _unreachable_endpoints;
 
+    seastar::shared_mutex _endpoint_update_mutex;
```

Comment:
As @asias mentioned earlier - we don't need a read-write lock because we're always locking exclusively. So a `semaphore` (with 1 unit) would be simpler and instead of using `with_lock(callback)`, you could do `auto lock = co_await get_units(semaphore, 1)` at the beginning of `replicate_live_endpoints_on_change`, resulting in less indentation and simpler code.

 But `shared_mutex` also works so I won't protest if you want to use it.